doc/requirements/sdid.rst

```diff
@@ -0,0 +1,261 @@
+Yet another error prevention tool is the customizability of individual settings. As mentioned previously a complete understanding of the complex meaning involved in setting values can't be programmed. However by giving each setting more character in the system to stand up on its own against mistakes (e.g. a temperature below absolute zero) can chip away at the burden of the user to need to understand the entirety.
+
+Each setting is intended to present a way of answering a question to the user from the system. For example many settings ask questions like what external code engine to utilize for advanced calculations, or what temperature to apply to a particular component. These questions are not open ended and as such usually have a set of rules surrounding their use like no temperatures below absolute zero, or only code engines specified by the following three strings are valid.
+
+The provided rules for how a setting behaves varies by subtype (i.e. Numeric settings possess valid maximums and minimums, whereas strings can be matched against a set of acceptable values). Given the extensibility of the setting system there is no limitation currently imposed by the system on what rules may apply. Only a basic set of rules currently exists but that has been found to be more than sufficient in modeling developer specifications for settings thus far.
+
+Most of these customized behaviors are directly linked with how the GUI representation of the settings restrict interactions with the user. For instance values outside of the allowable range for numeric settings will reject the change with an immediate reversion to the previously accepted value, or string settings with a restricted set of acceptable values will only display those in a drop down menu.
+
+
+Reactor Design
+--------------
+
+The physical hierarchy typical in a nuclear reactor is reflected in the design of the reactor package.
+It uses a `Composite Design Pattern <https://en.wikipedia.org/wiki/Composite_pattern>`_ to represent
+part-whole hierarchies. In other words, a Reactor is typically made of Assemblies, which are made of Blocks,
+which are made of Components, and so on. Requirements regarding the representation of a user-specified reactor
+are satisfied by the objects in this hierarchy.
+
+At each level of the hierarchy, the state can be found as a state variable called a *Parameter*. The parameter
+system is designed and implemented to satisfy the requirements related to storing and updating a dynamic state.
+
+
+Spatial Arrangements
+^^^^^^^^^^^^^^^^^^^^
+
+The :py:mod:`grids module  <armi.reactor.grids>` define where objects currently are in a regular, structured
+grid. In particular, *Assemblies* sit in the 2-D grid on the reactor and *Blocks* sit in 1-D grids on Assemblies.
+
+Setting and getting state variables
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The *state* is stored on component-level *parameters*. The design and implementation of this subpackage is fully described in
```

Comment:
Parameters are not only component-level. They also apply on block, assembly, core, etc.